annotations/quadriga.py

```diff
@@ -275,3 +257,280 @@ def parse_response(raw_response):
+        
+        # Create relation node
+        relation_id = f"rel-{relation.id}-{relation.created.isoformat()}"
+        if relation_id not in node_mapping:
+            node_id = get_node_id()
+            node_mapping[relation_id] = node_id
+            
+            source_uri = relation.occursIn.uri if hasattr(relation.occursIn, 'uri') else ""
+            nodes[node_id] = get_relation_node(user, relation.created, source_uri)
+            
+            # Add edges
+            if source_node_id:
+                edges.append({
+                    "source": node_id,
+                    "relation": "subject",
+                    "target": source_node_id
+                })
+            
+            if predicate_node_id:
+                edges.append({
+                    "source": node_id,
+                    "relation": "predicate",
+                    "target": predicate_node_id
+                })
+            
+            if object_node_id:
+                edges.append({
+                    "source": node_id,
+                    "relation": "object",
+                    "target": object_node_id
+                })
+                
+        return node_mapping[relation_id]
+    
+    def process_appellation(appellation):
+        appellation_id = f"app-{appellation.id}-{appellation.created.isoformat()}"
+        if appellation_id not in node_mapping:
+            node_id = get_node_id()
+            node_mapping[appellation_id] = node_id
+            
+            # Get the source URI from the document or default to empty
+            source_uri = appellation.occursIn.uri if hasattr(appellation.occursIn, 'uri') else ""
+            
+            # Build the node
+            concept_node = build_concept_node(appellation, user, appellation.created, source_uri)
+            nodes[node_id] = concept_node
+            
+        return node_mapping[appellation_id]
+    
+    # Process the top-level relation
+    top_relation = relationset.root
+    # Process the root relation, which will recursively process all nested relations
+    process_relation(top_relation)
+    
+    #############################################################################
+    # DEFAULT MAPPING SECTION
+    # 
+    # The default mapping identifies the "main" or top-level relationship in the graph.
+    # It has three components:
+    # 1. subject - The entity that the relation is about
+    # 2. predicate - The type of relationship
+    # 3. object - The entity that the subject relates to
+    #
+    # For Quadriga, this default mapping:
+    # - Provides an entry point to understand the graph
+    # - Identifies which relationship is most important
+    # - Allows immediate access to key parts without traversing the whole graph
+    #############################################################################
+    
+    # Initialize an empty default mapping dictionary
+    default_mapping = {}
```

Comment:
the default mapping should be coming from the template used to create a relation